nucliadb_texts2/src/writer.rs

```diff
@@ -296,6 +296,23 @@ impl TextWriterService {
             self.schema.status => resource.status as u64,
         );
 
+        let public_to_all_groups = resource.security.is_none()
+            || resource.security.as_ref().unwrap().access_groups.is_empty();
+        if public_to_all_groups {
+            base_doc.add_u64(self.schema.groups_public, 1_u64);
+        } else {
+            base_doc.add_u64(self.schema.groups_public, 0_u64);
+            for group_id in resource.security.as_ref().unwrap().access_groups.iter() {
```

Comment:
You are unwrapping here because the logic of the block ensures its safe, but there is a better way of encoding this that does not need that much logic tracking:
```rust
        let resource_security = resource.security.as_ref();
        if let Some(security_groups) = resource_security.filter(|i| !i.access_groups.is_empty()) {
            base_doc.add_u64(self.schema.groups_public, 0_u64);
            for group_id in security_groups.access_groups.iter() {
                let mut group_id_key = group_id.clone();
                if !group_id.starts_with('/') {
                    // Slash needs to be added to be compatible with tantivy facet fields
                    group_id_key = "/".to_string() + group_id;
                }
                let facet = Facet::from(group_id_key.as_str());
                base_doc.add_facet(self.schema.groups_with_access, facet)
            }
        } else {
            base_doc.add_u64(self.schema.groups_public, 1_u64);
        }
```